addons/web/static/src/views/fields/datetime/datetime_field.js

```diff
@@ -376,6 +376,17 @@ export const dateRangeField = {
         },
     ],
     supportedTypes: ["date", "datetime"],
+    isValid: (record, fieldname, fieldInfo) => {
+        if (fieldInfo.widget === "daterange") {
+            if (fieldInfo.options.end_date_field && (!record.data[fieldInfo.options.end_date_field] !== !record.data[fieldname])) {
+                return false;
+            }
+            if (fieldInfo.options.start_date_field && (!record.data[fieldInfo.options.start_date_field] !== !record.data[fieldname])) {
+                return false;
+            }
+        }
```

Comment:
what logic are you trying to enforce here? Shouldn't you check that the other date (end_date_field or start_date_field) is required, and if it is unset, return false? I don't understand why you compare the extra date with the "main" date (fieldname), and why you don't take the required attribute into account (`evaluateBooleanExpr(fieldInfo.required, record.evalContextWithVirtualIds)`)

Reply:
Hello @aab-odoo ,

Based on your suggestion, I have made the changes and it is working fine.
Regarding your question about comparing the extra date: when removing the main date. At that time, CSS should be applied. Previously, the behavior was such that when the main date was removed and the save button was clicked, the CSS would be applied.

video: [before commit](https://tinyurl.com/2mqk4ewe) and [after commit](https://tinyurl.com/2zwlbzp7)

Thank you for the suggestion.

Reply:
I still don't understand why the logic doesn't depend on the fact that the extra field is required or not. As the spec says:

> In the daterange widget, if the second date field is required and left empty ...

So I expect to see something like
```
if extra field is not set AND extra field is required:
    return false
```

Reply:
> I still don't understand why the logic doesn't depend on the fact that the extra field is required or not. As the spec says:
> 
> > In the daterange widget, if the second date field is required and left empty ...
> 
> So I expect to see something like
> 
> ```
> if extra field is not set AND extra field is required:
>     return false
> ```

Hello @aab-odoo ,

Based on your suggestion, I have made the changes and it is working fine.

Thank you for the suggestion.